### DISCUSSION TITLE
Block manager categories customization

### DISCUSSION THREAD

**Discussion by ddm50** [GrapesJS/grapesjs#5057] [Q&A]
I am looking to create custom component for the categories on block manager, I can't find any info
![image](https://user-images.githubusercontent.com/82258474/232937813-04190d32-1d71-4558-9fd0-ddd71b344bc1.png)

**Comment by ddm50** [comment]
....

**Comment by leonardolima99** [comment]
Hi @ddm50 , I did it this way. When you define the properties of the block, add a "category" with the name of the category you want. 

![Screenshot_20230424_104009](https://user-images.githubusercontent.com/28113634/234032168-7493f0d1-670c-45e9-8f69-6b9e61c84042.png)

To put other blocks in the same category, just set the same "category" value for them. Or you can do better, instead of just the category name, set an object with the category data, including whether it starts open or not:

`{ id: "columns", label: "Columns", open: false }`

 I hope I helped.

**Reply by ddm50** [reply]
I am looking to customize the entire thing there, not just name. Doesn't seem to be any way though... very annoying

**Reply by leonardolima99** [reply]
So you can customize things like font, size, colors, etc... The only way I've found is to overwrite the css classes. As for the functionality itself, I am also looking for this.

**Reply by ddm50** [reply]
> So you can customize things like font, size, colors, etc... The only way I've found is to overwrite the css classes. As for the functionality itself, I am also looking for this.

Yeah. It seems the only way is just DOM manipulation, replacing the entire content of gjs-title. Which works, but not ideal. I guess block customizability is fine enough given I can manipulate the look of blocks

**Reply by leonardolima99** [reply]
While we don't have the ideal solution, we do what we can with what we have available.

@artf  Any light at the end of this tunnel? Please...

**Reply by artf** [reply]
There is a guide on how to implement your own block manager UI [here](https://grapesjs.com/docs/modules/Blocks.html#customization)
